blaze/compute/sql.py

```diff
@@ -1205,6 +1208,43 @@ def compute_up(expr, data, **kwargs):
     return sa.sql.functions.char_length(data).label(expr._name)
 
 
+@compute_up.register(StrCat,
+                     (Select, ColumnElement, basestring),
+                     (Select, ColumnElement))
+@compute_up.register(StrCat,
+                     (Select, ColumnElement),
+                     basestring)
+def compute_up(expr, lhs, rhs, **kwargs):
+    # TODO: FIXME: Ensure the expr.lhs and expr.rhs come from the same table?
+    # This is not checked for any other binary column operation, so it's a
+    # pervasive issue throught Blaze's SQL implementation that should be
+    # addressed consistently.
+    if isinstance(lhs, Select):
+        assert len(lhs.c) == 1, (
+            'Select cannot have more than a single column when doing'
+            ' arithmetic, got %r' % lhs
+        )
+        lhs_col = first(lhs.inner_columns)
+    else:
+        lhs_col = lhs
+    if isinstance(rhs, Select):
+        assert len(rhs.c) == 1, (
+            'Select cannot have more than a single column when doing'
+            ' arithmetic, got %r' % rhs
+        )
+        rhs_col = first(rhs.inner_columns)
+    else:
+        rhs_col = rhs
+    if expr.sep:
+        sel = select((lhs_col + expr.sep + rhs_col).label(expr.lhs._name))
+    else:
+        sel = select((lhs_col + rhs_col).label(expr.lhs._name))
+    wheres = unify_wheres([lhs, rhs])
+    if wheres is not None:
```

Comment:
I think we need to use `reconstruct_select` here because if the comes from a select with something like a groupby or sort this information will be lost.